### ISSUE TITLE
false positive parent selector in `declaration-nested-properties` and `declaration-nested-properties-no-divided-groups` rule

### ISSUE THREAD

**Issue by evilebottnawi** [kristerkari/stylelint-scss#95]
Example
```shell
"scss/declaration-nested-properties": "never",
"scss/declaration-nested-properties-no-divided-groups": true
```

Code
```scss
.pagination-item-away {
        @media #{$xxs-and-down} {
                display: none;
        }

        &:not(.pagination-item-away-1, .pagination-item-away-2, .pagination-item-away-3, .pagination-item-away-4) {
                @media #{$md-only} {
                        display: none;
                }
        }

        &:not(.pagination-item-away-1, .pagination-item-away-2, .pagination-item-away-3) {
                @media #{$sm-only} {
                        display: none;
                }
        }

        &:not(.pagination-item-away-1, .pagination-item-away-2) {
                @media #{$xs-only} {
                        display: none;
                }
        }
}
```
Errors:
```shell
✖  Unexpected nested property "&"                                              scss/declaration-nested-properties                  
✖  Expected all nested properties of "&" namespace to be in one nested group   scss/declaration-nested-properties-no-divided-groups
✖  Unexpected nested property "&"                                              scss/declaration-nested-properties                  
✖  Expected all nested properties of "&" namespace to be in one nested group   scss/declaration-nested-properties-no-divided-groups
✖  Unexpected nested property "&"                                              scss/declaration-nested-properties                  
✖  Expected all nested properties of "&" namespace to be in one nested group   scss/declaration-nested-properties-no-divided-groups

```

**Comment by dryoma** [comment]
Fixed in 1.4.1.
